### ISSUE TITLE
Add a `CHANGELOG` file

### ISSUE THREAD

**Issue by aljo242** [tendermint/spn#916]
We should add a `CHANGELOG.md` that can easily track the changes we make across development.  This can be referenced when we make new releases (which we should probably also be doing).

A good example is the one found in [`ibc-go`](https://github.com/cosmos/ibc-go/blob/main/CHANGELOG.md)

**Comment by lubtd** [comment]
Yes, we will definitely add a changelog file. For now, we're not making releases but only tracking development with milestones, except for audit releases that are more symbolic
